runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp

```diff
@@ -0,0 +1,1434 @@
+	} else if (refClass == J9VMJAVALANGREFLECTMETHOD(vm)) {
+		J9JNIMethodID *methodID = vm->reflectFunctions.idFromMethodObject(currentThread, refObject);
+		vmindex = (jlong)methodID;
+		target = (jlong)methodID->method;
+
+		J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(methodID->method);
+
+		flags = romMethod->modifiers & CFR_METHOD_ACCESS_MASK;
+		flags |= MN_IS_METHOD;
+		if (J9_ARE_ANY_BITS_SET(methodID->vTableIndex, J9_JNI_MID_INTERFACE)) {
+			flags |= MH_REF_INVOKEINTERFACE << MN_REFERENCE_KIND_SHIFT;
+		} else if (J9_ARE_ANY_BITS_SET(romMethod->modifiers , J9AccStatic)) {
+			flags |= MH_REF_INVOKESTATIC << MN_REFERENCE_KIND_SHIFT;
+		} else if (J9_ARE_ANY_BITS_SET(romMethod->modifiers , J9AccFinal) || !J9ROMMETHOD_HAS_VTABLE(romMethod)) {
+			flags |= MH_REF_INVOKESPECIAL << MN_REFERENCE_KIND_SHIFT;
+		} else {
+			flags |= MH_REF_INVOKEVIRTUAL << MN_REFERENCE_KIND_SHIFT;
+		}
+
+		clazzObject = J9VMJAVALANGREFLECTMETHOD_DECLARINGCLASS(currentThread, refObject);
+	} else if (refClass == J9VMJAVALANGREFLECTCONSTRUCTOR(vm)) {
+		J9JNIMethodID *methodID = vm->reflectFunctions.idFromConstructorObject(currentThread, refObject);
+		vmindex = (jlong)methodID;
+		target = (jlong)methodID->method;
+
+		J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(methodID->method);
+
+		flags = romMethod->modifiers & CFR_METHOD_ACCESS_MASK;
+		flags |= MN_IS_CONSTRUCTOR | (MH_REF_INVOKESPECIAL << MN_REFERENCE_KIND_SHIFT);
+
+		clazzObject = J9VMJAVALANGREFLECTMETHOD_DECLARINGCLASS(currentThread, refObject);
+	} else {
+		vmFuncs->setCurrentExceptionUTF(currentThread, J9VMCONSTANTPOOL_JAVALANGILLEGALARGUMENTEXCEPTION, NULL);
+	}
+
+	if (!VM_VMHelpers::exceptionPending(currentThread)) {
+		J9VMJAVALANGINVOKEMEMBERNAME_SET_FLAGS(currentThread, membernameObject, flags);
+		J9VMJAVALANGINVOKEMEMBERNAME_SET_CLAZZ(currentThread, membernameObject, clazzObject);
+		J9OBJECT_U64_STORE(currentThread, membernameObject, vm->vmindexOffset, (U_64)vmindex);
+		J9OBJECT_U64_STORE(currentThread, membernameObject, vm->vmtargetOffset, (U_64)target);
+		Trc_JCL_java_lang_invoke_MethodHandleNatives_initImpl_setData(currentThread, flags, clazzObject, vmindex, target);
+	}
+}
+
+static char *
+sigForPrimitiveOrVoid(J9JavaVM *vm, J9Class *clazz)
+{
+	char result = 0;
+	if (clazz == vm->booleanReflectClass)
+		result = 'Z';
+	else if (clazz == vm->byteReflectClass)
+		result = 'B';
+	else if (clazz == vm->charReflectClass)
+		result = 'C';
+	else if (clazz == vm->shortReflectClass)
+		result = 'S';
+	else if (clazz == vm->intReflectClass)
+		result = 'I';
+	else if (clazz == vm->longReflectClass)
+		result = 'J';
+	else if (clazz == vm->floatReflectClass)
+		result = 'F';
+	else if (clazz == vm->doubleReflectClass)
+		result = 'D';
+	else if (clazz == vm->voidReflectClass)
+		result = 'V';
+
+	if (result) {
+		PORT_ACCESS_FROM_JAVAVM(vm);
+		char* signature = (char*)j9mem_allocate_memory(2 * sizeof(char*), OMRMEM_CATEGORY_VM);
```

Comment:
```suggestion
		char* signature = (char*)j9mem_allocate_memory(2, OMRMEM_CATEGORY_VM);
```
The current code is allocating two pointers worth of data (ie: 2*8 bytes on a 64bit system)